### DISCUSSION TITLE
wildcard argument using docker amazon/aws-cli

### DISCUSSION THREAD

**Discussion by torson** [aws/aws-cli#6922] [Q&A] [answered]
What I currently have set up is this alias in my .bashrc
```
alias aws='docker run --rm -v ~/.aws:/root/.aws -v $(pwd):/aws -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY -e AWS_REGION amazon/aws-cli'
```
And it works well except when I add some wildcard as argument:
```
aws cloudfront create-invalidation \
        --distribution-id ${CLOUDFRONT_DISTRIBUTION} \
        --paths '/*'
```

I get the response that I've invalidated these paths which is obviously my root folder listing:
```
{
    "Location": "https://cloudfront.amazonaws.com/....",
    "Invalidation": {
        "Id": "I2YK778ZANZPQS",
        "Status": "InProgress",
        "CreateTime": "2022-05-05T15:39:30.360000+00:00",
        "InvalidationBatch": {
            "Paths": {
                "Quantity": 24,
                "Items": [
                    "/lib",
                    "/lib64",
                    "/lib32",
                    "/boot",
                    "/libx32",
                    "/home",
                    "/sys",
                    "/run",
                    "/bin",
                    "/proc",
                    "/sbin",
                    "/var",
                    "/dev",
                    "/usr",
                    "/etc",
                    "/mnt",
                    "/opt",
                    "/snap",
                    "/cdrom",
                    "/media",
                    "/root",
                    "/srv",
                    "/tmp"
                ]
            },
            "CallerReference": "cli-1651765169-986767"
        }
    }
}
```

I've tried a few different ways to escape the `'/*'` but no success.

**Comment by torson** [comment, marked as answer]
Actually, this is happening only if I run the command within a bash script. The same command works well when running in a bash shell. How odd. 

Workaround for me was to install the awscli with pip3 which is currently still at v1.23.7 and run the cloudfront invalidation command with `env` which then uses the pip installed awscli instead of the bash alias one: 
```
env aws cloudfront create-invalidation \
        --distribution-id ${CLOUDFRONT_DISTRIBUTION} \
        --paths '/*'
```

**Reply by tim-finnigan** [reply]
Hi @torson I’m sure if I understand the issue but here is a link to our recommended installation methods: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html

Are you still blocked on this? If so it would help to see the debug logs from running that command.

**Reply by torson** [reply]
Thanks for the link, I might install it this way. 
I wasn't really blocked, the workaround I've mentioned works.
The issue is how within a bash script I use the bash alias command that runs the awscli docker container. Something in that chain is functioning in a way I don't understand in regards to bash globbing, so I just thought someone here might have some bash advanced skills. It doesn't look like it has anything to do with the awscli tool itself.

**Reply by torson** [reply]
I've now ran it with `--debug` and it shows that those paths are in the arguments :
```
2022-05-06 18:29:57,719 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--debug', 'cloudfront', 'create-invalidation', '--distribution-id', 'E2GL4CE753RUM8', '--paths', '/bin', '/boot', '/cdrom', '/dev', '/etc', '/home', '/lib', '/lib32', '/lib64', '/libx32', '/media', '/mnt', '/opt', '/proc', '/root', '/run', '/sbin', '/snap', '/srv', '/sys', '/tmp', '/usr', '/var']
```
So all good there from the context of awscli. It's the bash globbing that adds those paths when reading the `'/*'`
